MockMDS/src/main/java/io/mosip/mock/sbi/devicehelper/SBIDeviceHelper.java

```diff
@@ -818,8 +819,22 @@ private void loadKeys(String keyStoreFileName, String alias, String keystorePass
 	 * To be invoked in afterSuite
 	 * @param keystoreFilePath
 	 */
-	public static void evictKeys(String keystoreFilePath) {
-		privateKeyMap.entrySet().removeIf( e -> e.getKey().startsWith(keystoreFilePath));
-		certificateMap.entrySet().removeIf( e -> e.getKey().startsWith(keystoreFilePath));
+
+	
+	public static boolean evictKeys(String keystoreFilePath) {
+		boolean bKeyFound = false;
+		for (Entry<String, PrivateKey> entry : privateKeyMap.entrySet()) {

```

Comment:
Why is this check required?

Reply:
I think "removeIf" will return false in two cases
- removeIf operation failed
- If the key not present

Can we distinguish between those two.

Reply:
Why do you want to distinguish? Also, this is not thread-safe. With all these conditions you will have no assurance of thread safety. I don't think this logic is correct.

Reply:
if the removeif failed.. we want fail the DSL step (in before suite) and not execute other scenarios, because it will give spurious results.. because for every run we generate new device certificate and use it.., so if for some reason removeif is failed what we passing (device cert path) vs what we have in mds cache will different..so it will have issue..

Reply:
w.r.t. thread safety.. we can make it as synchronized  method..

Reply:
No that will further slow this down. I am still confused why you will need to know if the entry exist before removal or not? If it exists we remove if not we return and continue to load again so where is the problem?